solver/src/solver/http_solver.rs

```diff
@@ -202,9 +206,44 @@ impl HttpSolver {
                     },
                     mandatory: false,
                 };
-                (*index, uniswap)
+                (*index, pool_model)
             })
-            .collect()
+            .collect();
+        let balancer_cost = self.balancer_cost(gas_price);
+        let weighted_product_models: HashMap<_, AmmModel> = weighted_product_orders
+            .iter()
+            .map(|(index, amm)| {
+                let reserves = amm
+                    .reserves
+                    .iter()
+                    .map(|(token, state)| {
+                        (
+                            *token,
+                            PoolTokenData {
+                                balance: state.balance,
+                                // Probably shouldn't unwrap here.
+                                weight: bfp_to_big_decimal(state.weight).unwrap(),
```

Comment:
If we believe that the function cannot error then we should do the unwrap inside of it as an implementation detail so that the signature doesn't contain Result.

Reply:
I also really liked @vkgnosis's suggestion of doing:

> was to always store `BigRational` in structs and then use one of those `serde` attributes like `#[serde(with = "u256_decimal")]`

And would prefer this over doing the type conversion directly.

Reply:
OMG yes. I will do this if it actually works.

Reply:
Unfortunately, this results in a Serialization issue:

```
error[E0308]: mismatched types
  --> solver/src/solver/http_solver/model.rs:55:17
   |
55 | #[derive(Debug, Serialize)]
   |                 ^^^^^^^^^ expected struct `U256`, found struct `Ratio`
   ```
I suppose we have to manually implement this or something?

Reply:
Yes, you would have to implement your own `big_rational_decimal` formatter in the same way the `U256` one was implemented.

Reply:
This crate containing BigRational (https://docs.rs/num/0.4.0/) doesn't appear to have a `serde_with` thingy

Reply:
Ahh I see now, I wasn't aware we had written this for ourselves.

Reply:
At the end of the day I am still going to have to write a string representation of this. Here is my most recent approach

```rust
    let float_rep = value.numer().to_f64().unwrap() / value.denom().to_f64().unwrap();
    serializer.serialize_str(&float_rep.to_string())
```

I mean there are so many ways to do this, but we will have to do it.

Reply:
I will create a separate PR for this BTW.